opm/material/fluidsystems/BlackOilFluidSystem.hpp

```diff
@@ -994,6 +994,41 @@ class BlackOil : public BaseFluidSystem<Scalar, BlackOil<Scalar> >
     }
 
     /*!
+     * \brief Returns the bubble point pressure $P_b$ using the current gas saturation in the fluid phase (Rs)
+     */
+    template <class FluidState, class LhsEval = typename FluidState::Scalar>
+    static LhsEval bubblePointPressure(const FluidState& fluidState,
+                                       unsigned regionIdx)
+    {
+        assert(0 <= regionIdx && regionIdx <= numRegions());
+
+        typedef Opm::MathToolbox<typename FluidState::Scalar> FsToolbox;
+
+        const auto& T = FsToolbox::template decay<LhsEval>(fluidState.temperature(oilPhaseIdx));
+        const auto& Rs = FsToolbox::template decay<LhsEval>(fluidState.Rs());
+
+        return oilPvt_->saturationPressure(regionIdx, T, Rs);
+    }
+
+
+    /*!
+     * \brief Returns the dew point pressure $P_d$ using the current fluid saturation in the gas phase (Rv)
```

Comment:
I'm not in complete control of this, but isn't it sufficient with the Rs/Rv factors and pressure to determine when the gas/oil bubbles/"dews"?

Perhaps @atgeirr also can comment and see if I've misunderstood something here?